MANTA-5203 rebalancer should not automatically send assignments for every shark update

rjloura opened the pull request:
MANTA-5203 rebalancer should not automatically send assignments for every shark update
MANTA-5240 rebalancer should not allow old assignments to linger post MANTA-5203
MANTA-5239 rebalancer post and assignment generator threads could use better logging

rhb2 commented:
Looks to me like it was a bit more than a variable rename.  You're no longer hanging the maximum number of tasks allowed off of the job.  Instead, it's obtained from the config options which (if I'm understanding correctly hangs off of the evac job structure anyway, so you've cut down on retaining redundant information).  Let me know if I'm understanding the motivation for that correctly or not.  It may be worth just having a method on the evac job that can give us that information in the event that it moves around more in the future. :)

rjloura commented:
> Looks to me like it was a bit more than a variable rename. You're no longer hanging the maximum number of tasks allowed off of the job. Instead, it's obtained from the config options which (if I'm understanding correctly hangs off of the evac job structure anyway, so you've cut down on retaining redundant information). Let me know if I'm understanding the motivation for that correctly or not. It may be worth just having a method on the evac job that can give us that information in the event that it moves around more in the future. :)

I added "max_tasks_per_assignment" originally knowing that I wanted to eventually make it configurable.  When I did, I added "max_assignment_size" and totally forgot that I created "max_tasks_per_assignment".  

Essentially once I got around to actually using the variable I had renamed it, and the old version amounted to cuft.  It is really an oversight on my part.

rhb2 commented:
On second thought, unless we can come up with a way of ensuring that this tunable does not get wiped from the config file when SAPI tunable changes, I think it might be a bit dangerous to introduce this knob -- or at least introduce it as something that can be consumed from the config file.

There are at least two options that I can think of off the top of my head:
* Throw it in the template file that config agent uses to generate the config
* Perhaps supply it as a command line argument when the service starts

There might be other ways beyond the scope of my knowledge, but those are the two primary mitigations that come to mind.  Either of these changes are pretty minor, so it probably wouldn't be a big deal to couple one with the introduction of this knob.
